micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/correlationid/CorrelationIdAspect.java

```diff
@@ -80,6 +90,17 @@ public Object call() throws Exception {
         });
     }
 
+    @Around("anyControllerOrRestControllerWithPublicWebAsyncTaskMethod()")
+    public Object wrapWebAsyncTaskWithCorrelationId(ProceedingJoinPoint pjp) throws Throwable {
+        final WebAsyncTask webAsyncTask = (WebAsyncTask) pjp.proceed();
+        log.debug("Wrapping webAsyncTask with correlation id [" + CorrelationIdHolder.get() + "]");
+
+        Field callableField = WebAsyncTask.class.getDeclaredField("callable");
```

Comment:
I agree it'd be better to use the API only if possible but how do you set the onTimeout and onCompletion callbacks on the new WebAsyncTask?